[86c04hbeu][data-table] data table sort column properties

ilyabrower opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Motivation and Context
I've disabled changing sorting columns width by default.
I've added `changeSortSize` to enable this logic on columns and `sortSizeRecalculation` to set by which columns we should recalculate width of table. By default we using column width maximum width.
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How has this been tested?
Manually
<!--- Please describe in detail how you tested your changes. -->
<!--- For example: -->
<!--- I have added unit tests -->
<!--- I have added Voice Over tests -->
<!--- Code cannot be tested automatically so I have tested it only manually -->

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue).
- [X] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Nice improve.

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [X] My code follows the code style of this project.
- [X] I have updated the documentation accordingly or it's not required.
- [X] Unit tests are not broken.
- [X] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [ ] I have added new unit tests on added of fixed functionality.

sheila-semrush reviewed (changes requested):
1. A line break in the "KD, %" column appeared in Firefox:
<img width="725" alt="image" src="https://github.com/user-attachments/assets/8e8fffb7-a2cd-4d17-b1a1-b44b945ca1f4">

#

2. Also discussed to slightly change the examples: make the "Vol" column noticeably smaller than "Keyword", and in the last example assign the sortSizeRecalculation prop to the "Vol", so it's obvious that we can take space from a column that's not the largest.

ilyabrower commented:
> 1. A line break in the "KD, %" column appeared in Firefox:
> 
> <img alt="image" width="725" src="https://private-user-images.githubusercontent.com/166654065/364762870-8e8fffb7-a2cd-4d17-b1a1-b44b945ca1f4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2MTA1MjksIm5iZiI6MTcyNTYxMDIyOSwicGF0aCI6Ii8xNjY2NTQwNjUvMzY0NzYyODcwLThlOGZmZmI3LWEyY2QtNGQxNy1iMWExLWI0NGI5NDVjYTFmNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwNlQwODEwMjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01YWJmODJjNDY3NmYxN2U5MWVlZDBhNWM4OWZkZTFkMjc2ZjA0MzhkNDg3YmZjOGJjZDgxZWE2ODQ1OWRlMjMxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.bRL3MrQPgQLr7iWXFCOEwK8NQESUJEsO2gKKNvoOpSQ">
> 2. Also discussed to slightly change the examples: make the "Vol" column noticeably smaller than "Keyword", and in the last example assign the sortSizeRecalculation prop to the "Vol", so it's obvious that we can take space from a column that's not the largest.

Fixed the first one and changed examples

sheila-semrush reviewed (approved):
All good!
